src/Diagnostics.DataProviders/DataProviderConfigurations/KustoDataProviderConfiguration.cs

```diff
@@ -222,6 +230,19 @@ public override void PostInitialize()
                     FailoverClusterNameCollection.TryAdd(clusterNameGroupingParts[iterator], clusterFailoverGroupingParts[iterator]);
                 }
             }
+
+            int numOfOverridableExceptions = config != null? config.GetSection("Kusto").GetSection("Retry").GetSection("OverridableExceptionsToRetryAgainstLeaderCluster").GetChildren().ToList().Count() : 0;
```

Comment:
Instead of counting the elements in the array and then looping through them, see if you can make use of foreach